src/torrent_info.cpp

```diff
@@ -1747,6 +1747,10 @@ namespace {
 		e.source = source;
 		m_urls.push_back(e);
 
+#if TORRENT_USE_I2P
+		if (is_i2p_url(e.url)) m_flags |= i2p;
```

Comment:
I believe it should be pretty easy to add a unit test for this, which I think would be good. Also, what do you think of checking `m_flags & i2p` as the first condition, to avoid calling `is_i2p_url()` when the flag has already been set.

Reply:
I made more tests and found that added check to `torrent_info::add_tracker` fixes #7260 only when fresh torrent is added.
When torrent is started from resume data, `torrent::is_i2p` function returns `false` because `m_torrent_file` have no `m_urls`.
Because of it, I propose to change line
https://github.com/arvidn/libtorrent/blob/32e36984c944e85eacc57c381db4605d6636aa84/src/torrent.cpp#L3148
to `if (is_i2p_url(req.url))` instead.
@arvidn can you check if it is better way to fix #7260 ?

Reply:
when I made the comment I mistakenly thought this was done in a loop.
I think setting the flag here makes more sense than checking tracker URLs in the `is_i2p()` function